homeassistant/components/bluetooth/match.py

```diff
@@ -237,10 +237,12 @@ def build(self) -> None:
     def match(self, service_info: BluetoothServiceInfoBleak) -> list[_T]:
         """Check for a match."""
         matches = []
-        if service_info.name and len(service_info.name) >= LOCAL_NAME_MIN_MATCH_LENGTH:
-            for matcher in self.local_name.get(
-                service_info.name[:LOCAL_NAME_MIN_MATCH_LENGTH], []
-            ):
+        if (name := service_info.name) and (
+            local_name_matchers := self.local_name.get(
+                name[:LOCAL_NAME_MIN_MATCH_LENGTH:]
```

Comment:
This 2nd `:` is not needed right?

Reply:
Its not needed. For some reason I was thinking it would raise `IndexError` if the string wasn't 3 long but it works just fine.